add docs for str lib and changed some function names to underscore separated names (#84)
arr-ai/arrai: docs/std-str.md

# str library

str library contains functions that are used for string manipulations.

## concat

> `//.str.concat` concatenates a list of strings. It takes one argument which is

Reviewer: ```suggestion
`//.str.concat(strings)` concatenates a list of strings. It takes one argument which is
```

Repeat this for all definitions.

a list string. It returns a string.

> ### examples

Reviewer: `####` for easier differentiation

```text
//.str.concat(["ba", "na", "na"])

"banana"

> ```

Reviewer: ```suggestion
- `//.str.concat(["ba", "na", "na"]) = "banana"`
```

Reviewer: Use the same pattern everywhere.

Reviewer: Alternatively, use tables:

| expression | equals |
|:-|:-|
| `//.str.concat(["ba", "na", "na"])` | `"banana"` |

```
| expression | equals |
|:-|:-|
| `//.str.concat(["ba", "na", "na"])` | `"banana"` |
```

## contains

`//.str.contains` checks whether a substring is contained in a string. It takes
two arguments which are the string and the substring you check. It returns a
boolean.

### examples

```text
//.str.contains("the full string which has substring", "substring")
true

//.str.contains("just some random sentence", "microwave")

> {}

Reviewer: `... = false`.

```

## sub

> `//.str.sub` returns a string whose occurences of `old` string are replaced

Reviewer: ```suggestion
`//.str.sub(s, old, new)` returns a string whose occurrences of `old` string are replaced
```
